src/Gemstone.IO/Collections/FileBackedLookupTable.cs

```diff
@@ -1727,74 +1723,139 @@ static FileBackedLookupTable()
 
     // Static Methods
 
-    private static Action<Stream, T>? GetWriteMethod<T>()
+    private static Action<Stream, object>? GetWriteMethod(Type type)
     {
-        Type type = typeof(T);
-        MethodInfo? method = type.GetMethod("WriteTo", Flags, null, s_types, null);
+        MethodInfo? method = type.GetMethod("WriteTo", s_instanceFlags, null, s_types, null);
 
         if (method is not null)
-        {
-            Action<T, Stream> action = (Action<T, Stream>)Delegate.CreateDelegate(typeof(Action<T, Stream>), method);
-            return (stream, obj) => action(obj, stream);
-        }
+            return (stream, obj) => method.Invoke(obj, [stream]);
 
-        Action<BinaryWriter, T>? binaryWriterAction = GetBinaryWriterMethod<T>();
+        Action<BinaryWriter, object>? binaryWriterAction = GetBinaryWriterMethod(type);
 
         if (binaryWriterAction is null)
             return null;
 
         return (stream, obj) =>
         {
-            using BinaryWriter writer = new BinaryWriter(stream, Encoding.UTF8, true);
+            using BinaryWriter writer = new(stream, Encoding.UTF8, true);
             binaryWriterAction(writer, obj);
         };
     }
 
-    private static Func<Stream, T>? GetReadMethod<T>()
+    private static Action<Stream, T>? GetWriteMethod<T>()
     {
         Type type = typeof(T);
-        ConstructorInfo? constructor = type.GetConstructor(Flags, null, s_types, null);
+
+        if (!IsListType(type))
```

Comment:
I guess we can't add custom serialization directly to an implementation of `IList<>`. We could create a serializable wrapper type, though. Just making a note of this now; in case it ever comes up, maybe we'll remember.

Reply:
That's exactly what drove this change - a "Collection" class

Reply:
I meant to say that something like this would not work as you might expect. It will use the serialization mechanism for the underlying type instead of the collection itself.
```c#
public class MyCollection<T> : IList<T>
{
    public MyCollection(Stream stream) { ... }
    public void WriteTo(Stream stream) { ... }
}
```